src/redux/sagas/motions/managePermissionsMotion.ts

```diff
@@ -25,8 +27,170 @@ import {
   initiateTransaction,
   uploadAnnotation,
   createActionMetadataInDB,
+  getPermissionProofsLocal,
 } from '../utils/index.ts';
 
+function* getCreateMotionParams({
+  roles,
+  authority,
+  isMultiSig,
+  colonyAddress,
+  motionDomainId,
+  colonyRoles,
+  colonyDomains,
+  userAddress,
+  domainId,
+  batchKey,
+  metaId,
+}: {
+  roles: Record<ColonyRole, boolean>;
+  authority: Authority;
+  isMultiSig: boolean;
+  colonyAddress: string;
+  motionDomainId: number;
+  colonyRoles: ColonyRoleFragment[];
+  colonyDomains: Domain[];
+  userAddress: string;
+  domainId: number;
+  batchKey: TRANSACTION_METHODS;
+  metaId: string;
+}) {
+  const colonyManager = yield getColonyManager();
+
+  const colonyClient = yield colonyManager.getClient(
+    ClientType.ColonyClient,
+    colonyAddress,
+  );
+
+  const roleArray = Object.values(roles).reverse();
+  roleArray.splice(2, 0, false);
+
+  let roleBitmask = '';
+
+  roleArray.forEach((role) => {
+    roleBitmask += role ? '1' : '0';
+  });
+
+  const hexString = hexlify(parseInt(roleBitmask, 2));
+  const zeroPadHexString = hexZeroPad(hexString, 32);
+
+  let altTarget: string;
+
+  switch (authority) {
+    case Authority.ViaMultiSig: {
+      const multiSigClient = yield colonyManager.getClient(
+        ClientType.MultisigPermissionsClient,
+        colonyAddress,
+      );
+      altTarget = multiSigClient.address;
+      break;
+    }
+    case Authority.Own:
+    default: {
+      altTarget = isMultiSig ? colonyAddress : ADDRESS_ZERO;
+      break;
+    }
+  }
+
+  let initiatorAddress: string;
+
+  if (isMultiSig) {
+    initiatorAddress = yield colonyClient.signer.getAddress();
+  } else {
+    const votingReputationClient = yield colonyManager.getClient(
+      ClientType.VotingReputationClient,
+      colonyAddress,
+    );
+
+    initiatorAddress = votingReputationClient.address;
+  }
+
+  const requiredRoles =
+    motionDomainId === Id.RootDomain
+      ? PERMISSIONS_NEEDED_FOR_ACTION.ManagePermissionsInRootDomain
+      : PERMISSIONS_NEEDED_FOR_ACTION.ManagePermissionsInSubDomain;
+
+  const [permissionDomainId, childSkillIndex] = yield call(
+    getPermissionProofsLocal,
+    {
+      networkClient: colonyClient.networkClient,
+      colonyRoles,
+      colonyDomains,
+      requiredDomainId: motionDomainId,
+      requiredColonyRole: requiredRoles,
+      permissionAddress: initiatorAddress,
+      isMultiSig,
+    },
+  );
+
+  const encodedAction = colonyClient.interface.encodeFunctionData(
+    'setUserRoles',
+    [
+      permissionDomainId,
+      childSkillIndex,
+      userAddress,
+      domainId,
+      zeroPadHexString,
+    ],
+  );
+
+  if (isMultiSig) {
+    return {
+      context: ClientType.MultisigPermissionsClient,
+      methodName: 'createMotion',
+      identifier: colonyAddress,
+      params: [Id.RootDomain, childSkillIndex, [altTarget], [encodedAction]],
```

Comment:
Note to self: Check it works okay to use `Id.RootDomain` here and that the multi-sig motion is created in the correct domain.